ipyvolume/embed.py

```diff
@@ -1,202 +1,199 @@
 import os
-import json
+import io
+import zipfile
 import shutil
-import ipywidgets
+from ipywidgets import embed as wembed
 import ipyvolume
-from base64 import standard_b64encode
+from ipyvolume.utils import download_to_file, download_to_bytes
 
-template = """<!DOCTYPE html>
+html_template = u"""<!DOCTYPE html>
 <html lang="en">
 <head>
     <meta charset="UTF-8">
     <title>{title}</title>
-{extra_script_head}
+    {extra_script_head}
 </head>
 <body>
 {body_pre}
-
-<script src="{embed_url}"></script>
-<script type="application/vnd.jupyter.widget-state+json">
-{json_data}
-</script>
-{widget_views}
-
+{snippet}
 {body_post}
 </body>
 </html>
 """
 
-widget_view_template = """<script type="application/vnd.jupyter.widget-view+json">
-{{
-    "model_id": "{model_id}"
-}}
-</script>"""
 
+def save_ipyvolumejs(folderpath="", version=ipyvolume._version.__version_js__, devmode=False):
+    """ output the ipyvolume javascript to a local file
+    
+    :type folderpath: str
+    :type version: str
+    :type devmode: bool
 
-template_external = """<!DOCTYPE html>
-<html lang="en">
-<head>
-    <meta charset="UTF-8">
-    <title>{title}</title>
-{extra_script_head}
-</head>
-<body>
-{body_pre}
+    """
+    url = "https://unpkg.com/ipyvolume@{version}/dist/index.js".format(version=version)
+    pyv_filename = 'ipyvolume_v{version}.js'.format(version=version)
+    pyv_filepath = os.path.join(folderpath, pyv_filename)
 
-<script src="https://unpkg.com/jupyter-js-widgets@~2.0.20/dist/embed.js"></script>
-<script type="javascript">
-widget_data = {json_data}
-</script>
-{widget_views}
+    devfile = os.path.join(os.path.abspath(ipyvolume.__path__[0]), "..", "js", "dist", "index.js")
+    if devmode and os.path.exists(devfile):
+        if folderpath and not os.path.exists(folderpath):
+            os.makedirs(folderpath)
+        shutil.copy(devfile, pyv_filepath)
+    else:
+        download_to_file(url, pyv_filepath)
 
-{body_post}
-</body>
-</html>
-"""
+    three_filename = 'three_v{version}.js'.format(version=version)
```

Comment:
threejs has it's own version, this might be confusing. The threejs version also won't change often, so please just keep it three.js, or we should add the threejs version somewhere in this file (you can find the version in js/modules.json.